serve/mlc_serve/engine/staging_engine.py

```diff
@@ -218,31 +221,36 @@ def step(self) -> InferenceStepResult:
                 gen_seq = state.generation_sequences[seq_output.id.sequence_index]
                 new_token_ids = seq_output.new_tokens
 
-                delta = update_sequence(
-                    gen_seq,
-                    new_token_ids,
-                    state.prompt_token_ids,
-                    self.tokenizer,
-                    state.stopping_criteria,
-                )
+                if new_token_ids:
+                    delta = update_sequence(
+                        gen_seq,
+                        new_token_ids,
+                        state.prompt_token_ids,
+                        self.tokenizer,
+                        state.stopping_criteria,
+                    )
+                else:
+                    delta = None
```

Comment:
@sunggg This change might fix the repeated token issue. If `new_token_ids` is empty, we return an empty delta.